### PR TITLE
feat(buttongroup)!: migrate to S2

### PR DESCRIPTION
**Pull request by jenndiaz**
## Description

BREAKING CHANGE 

Migrations the button group component to S2 
- updates tokens for vertical and horizontal spacing with in the buttongroup component 
- includes size small with in Chromatic 

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] In storybook, buttongroup gap spacing match S2 token updates @mdt2 
- [x] In the docs site, buttongroup gap spacing match S2 token updates @mdt2 
- [x] VRTs now also display a small buttongroup in addition to the default size @mdt2 

- [x] design validation  @jenndiaz 

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [ ] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

**Review by mdt2 — APPROVED** [review]
Nice work. Aligns with the spec tokens. Thanks for adding the Chromatic coverage for small 🎉

File: components/buttongroup/index.css [hunk 1]
```diff
@@ -10,29 +10,17 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-.spectrum-ButtonGroup {
-  --spectrum-buttongroup-spacing-horizontal: var(--spectrum-spacing-300);
-  --spectrum-buttongroup-spacing-vertical: var(--spectrum-spacing-300);
-}
-
-.spectrum-ButtonGroup--sizeS {
+.spectrum-ButtonGroup,
+.spectrum-ButtonGroup--sizeM,
+.spectrum-ButtonGroup--sizeL,
+.spectrum-ButtonGroup--sizeXL {
```
Comment (jawinn) [inline components/buttongroup/index.css:16]: Could we remove these three classes, since only small is different than the default?

**Review by jawinn — APPROVED** [review]
LGTM. I tested the Chromatic template as well.

File: components/buttongroup/index.css (hunk 1 above)
Comment (jenndiaz) [reply components/buttongroup/index.css]: yes, I've removed them and updated our docs and migration guide to reflect them being removed.

File: components/buttongroup/metadata/buttongroup.yml [hunk 2]
```diff
@@ -1,14 +1,19 @@
 name: Button group
 SpectrumSiteSlug: https://spectrum.adobe.com/page/button/
 sections:
+  - name: Migration Guide
+    description: |
+      ### 1/24/2024 - Version 7.0.0
```
Comment (jenndiaz) [inline components/buttongroup/metadata/buttongroup.yml:6]: @pfulton 
Curious your thoughts are adding the date and version number here? Also, am I understanding correctly that as a breaking change the version number would bump from 6.2.0 to 7.0.0?

Comment (mdt2) [reply components/buttongroup/metadata/buttongroup.yml:6]: I love this! I'm not convinced that the date is needed, but including the version is important for future-proofing this and being able to determine what may or may not be relevant in the docs.

Comment (pfulton) [reply components/buttongroup/metadata/buttongroup.yml:6]: Yeah, this is nice! Thank you for adding it!

**Review by pfulton — APPROVED** [review]

**Pull request merged**